tests/typecheck_typeshed.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+"""Run mypy on the "tests" and "scripts" directories."""
+from __future__ import annotations
+
+import argparse
+import subprocess
+import sys
+from itertools import product
+from typing_extensions import TypeAlias
+
+from colors import colored, print_error
+
+ReturnCode: TypeAlias = int
+
+SUPPORTED_PLATFORMS = ("linux", "darwin", "win32")
+SUPPORTED_VERSIONS = ("3.11", "3.10", "3.9")
+DIRECTORIES_TO_TEST = ("scripts", "tests")
+
+parser = argparse.ArgumentParser(description="Run mypy on typeshed's own code in the `scripts` and `tests` directories.")
+parser.add_argument(
+    "--platform",
+    choices=SUPPORTED_PLATFORMS,
+    nargs="*",
+    action="extend",
+    help="Run mypy for certain OS platforms (defaults to sys.platform)",
+)
+parser.add_argument(
+    "-p",
+    "--python-version",
+    choices=SUPPORTED_VERSIONS,
+    nargs="*",
+    action="extend",
+    help="Run mypy for certain Python versions (defaults to sys.version_info[:2])",
+)
+parser.add_argument(
+    "-d",
+    "--dir",
+    choices=DIRECTORIES_TO_TEST,
+    nargs="*",
+    action="extend",
+    help=f"Test only these top-level typeshed directories (defaults to {DIRECTORIES_TO_TEST!r})",
+)
+
+
+def run_mypy_as_subprocess(directory: str, platform: str, version: str) -> ReturnCode:
+    command = [
+        sys.executable,
+        "-m",
+        "mypy",
+        directory,
+        "--platform",
+        platform,
+        "--python-version",
+        version,
+        "--strict",
+        "--show-traceback",
+        "--show-error-codes",
+        "--no-error-summary",
+        "--enable-error-code",
+        "ignore-without-code",
+        "--namespace-packages",
+    ]
+    if directory == "tests":
+        if platform == "win32":
+            command.extend(["--exclude", "tests/pytype_test.py"])
+    else:
+        command.append("--ignore-missing-imports")
```

Comment:
Side note: We should probably add the types for the imports we use to tests-requirements. (In a different PR.)

Reply:
Or maybe `script-requirements`/`stubsabot-requirements`? (Most contributors aren't going to need to run stubsabot manually, after all :)

Reply:
I think it's easier to just put all the requirements into one file. It won't hurt much to install these extra requirements, but it's a bit lower of a mental burden for when you need those.

Reply:
Looking at it again, it seems as though stubsabot's additional requirements are pretty minimal anyway, so it probably doesn't hurt -- you're right!